runtime/onert/backend/train/KernelGenerator.cc

```diff
@@ -252,14 +241,22 @@ void KernelGenerator::visit(const ir::train::operation::Loss &node)
   auto y_true_tensor = _tensor_reg->getPortableTensor(y_true_index);
 
   auto back_prop_y_pred_tensor = _tensor_reg->getBackPropTensor(y_pred_index);
-  auto fn = std::make_unique<ops::LossLayer>();
 
-  fn->configure(y_pred_tensor, y_true_tensor, output_tensor, back_prop_y_pred_tensor,
-                convertLossType(node.param().op_type));
+  auto op_type = node.param().op_type;
 
-  _return_fn = std::move(fn);
+  std::unique_ptr<ops::LossLayer> fn;
+  switch (op_type)
+  {
+    case ir::operation::Loss::Type::MEAN_SQUARED_ERROR:
+      fn = std::make_unique<ops::LossMeanSquaredErrorLayer>();
+      dynamic_cast<ops::LossMeanSquaredErrorLayer *>(fn.get())->configure(
```

Comment:
@jyoungyun `dynamic_cast` has performance issue. I prefer not to use `dynamic_cast` except that it is not avoidable. I don't think much, but is it possible to remove `dynamic_cast` here?

Reply:
I agree with @glistening . And there is no reason to use `dynamic_cast`.